src/sentry/models/commitauthor.py

```diff
@@ -32,12 +38,11 @@ class Meta:
 
     __repr__ = sane_repr("organization_id", "email", "name")
 
-    def find_users(self):
-        from sentry.models import User
+    def find_users(self) -> List["APIUser"]:
+        from sentry.services.hybrid_cloud.user import user_service
 
-        return User.objects.filter(
-            emails__email__iexact=self.email,
-            emails__is_verified=True,
-            sentry_orgmember_set__organization=self.organization_id,
-            is_active=True,
-        )
+        users = user_service.get_many_by_email(self.email)
+        group_memberships = OrganizationMember.objects.filter(
+            organization_id=self.organization_id,
```

Comment:
Should this be filtering by user_id. This query will attempt to load all members in the org (which could be a few hundred rows)? Previously the user query was filtering by email.

Reply:
Good catch. It definitely should filter by the list of users fetched in `get_many_by_email`

Reply:
I addressed this in latest commit.